pkg/trace/api/api_test.go

```diff
@@ -748,27 +769,47 @@ func TestClientComputedTopLevel(t *testing.T) {
 			if on {
 				req.Header.Set(headerComputedTopLevel, "yes")
 			}
+			errs := make(chan error, 2)
 			var wg sync.WaitGroup
 			wg.Add(1)
 			go func() {
 				defer wg.Done()
 				resp, err := http.DefaultClient.Do(req)
 				if err != nil {
-					t.Fatal(err)
+					errs <- err
+					return
 				}
 				if resp.StatusCode != 200 {
-					t.Fatal(resp.StatusCode)
+					errs <- fmt.Errorf("%d", resp.StatusCode)
+					return
 				}
 			}()
-			timeout := time.After(time.Second)
-			for {
-				select {
-				case p := <-rcv.out:
-					assert.Equal(t, p.ClientComputedTopLevel, on)
-					wg.Wait()
-					return
-				case <-timeout:
-					t.Fatal("no output")
+			var wg2 sync.WaitGroup
+			wg2.Add(1)
```

Comment:
> Rather than converting (*T).Fatal to (*T).Error, I choosed to keep the (*T).Fatal but 

If the latter is an option (using `(*T).Error`). I think I'd prefer that to this. What is going on here now is overly complicated. Can we just use `T.Error` instead?

Reply:
Sure!
I’ve just updated `api_test.go` to make it use `(*T).Error` instead of `(*T).Fatal` and to remove the plumbing code I added.

For `api_oom_test.go`, I stuck with the move of `(*T).Fatal` to keep the same intended behaviour of the test, i.e. have the test abort as soon as the first `http.Post` in any of the 50 goroutines returns an error.
If we replace `(*T).Fatal` by `(*T).Error` in `api_oom_test.go`, it would make the test wait for all the 50 goroutines to complete even if one returned an error very early.
WDYT ? Should we keep `api_oom_test.go` like this or would you prefer to also just replace the `(*T).Fatal` by `(*T).Error` ?

Reply:
It's perfect. Thank you!